Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -115,12 +115,22 @@ public async Task<HttpResponseMessage> CheckSetupAsync()
         [Route("/v1/teams/{aadGroupId}/update")]
         public async Task<ActionResult> UpdateTeam([FromRoute] string aadGroupId)
         {
+            var request = this.Request;
+            var requestHeaders = request.Headers;
+            Microsoft.Extensions.Primitives.StringValues passThroughValue = string.Empty;
+
             this.telemetryClient.TrackTrace("IncomingRequest, starts for method: UpdateTeam - " + DateTime.Now.ToString(CultureInfo.InvariantCulture));
 
             // Step 1 - Obtain the secret from the database.
             var configurationEntities = await this.configurationProvider.GetConfigurationsAsync().ConfigureAwait(false);
             var configurationEntity = configurationEntities?.FirstOrDefault();
 
+            // Check whether passthrough header is present and is equal to workforce integration id.
+            var isPassthroughHeaderValid = requestHeaders.TryGetValue("X-MS-WFMPassthrough", out passThroughValue) ?
```

Comment:
Rename this variable as the current name is very misleading. Any passthrough header value is technically **valid**. What you're determining here is whether this notification is a result of a Graph call from your integration.